bindings/python/scripts/parse.py

```diff
@@ -56,7 +30,7 @@ def valid_children(node):
     for child in cursor.get_children():
         child_node = {"cursor": child, "filename": filename, "depth": depth + 1}
         # Check if the child belongs to the file
-        if is_node_in_this_file(child_node):
+        if child.location.file is not None and child.location.file.name == filename:
```

Comment:
What about just `child.location.file` for truthiness? It's stronger in the sense that file as empty string aka `''` will also not be allowed